src/containers/PersonContainer/PersonRoles.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { FC } from "react";
+import { Link } from "react-router-dom";
+
+import Role from "../../models/Role";
+
+import Details from "../../components/Shared/Details";
+import H2 from "../../components/Shared/H2";
+import Ul from "../../components/Shared/Ul";
+
+import { fontSizes } from "../../styles/fonts";
+
+interface PersonRolesProps {
+  /** The person's councilmember roles */
+  councilMemberRoles: Role[];
+  /** The person's non councilmember roles, partitioned by concilmember roles' date range
+   * and whether the role is active or inactive.
+   * The first list of roles are active, while the second list of roles are inactive.
+   */
+  nonCouncilMemberRoles: Record<string, Role[][]>;
+}
+
+const PersonRoles: FC<PersonRolesProps> = ({
+  councilMemberRoles,
+  nonCouncilMemberRoles,
+}: PersonRolesProps) => {
+  // councilMemberRoles is guaranteed to have > 0 roles
+  return (
+    <div>
+      <H2 hasBorderBottom={true} className="mzp-u-title-xs">
+        Terms
+      </H2>
+      <Ul gap={16}>
+        {councilMemberRoles.map((role) => {
+          const isCurrentRole = !role.end_datetime || role.end_datetime > new Date();
+          return (
+            <li key={role.id}>
+              <Details
+                defaultOpen={isCurrentRole}
+                summaryContent={
+                  <span style={{ fontSize: fontSizes.font_size_6 }}>
+                    <strong>{`${role.title}: `}</strong>{" "}
+                    {`${role.seat?.name} // ${
+                      role.seat?.electoral_area
+                    } (${role.start_datetime.toLocaleDateString()} - ${
+                      role.end_datetime ? role.end_datetime.toLocaleDateString() : ""
+                    })`}
+                  </span>
+                }
+                hiddenContent={
+                  <Ul gap={8}>
+                    {nonCouncilMemberRoles[role.id][0].length > 0 && (
+                      <li>
+                        <Details
+                          defaultOpen={isCurrentRole}
+                          summaryContent={<span>Committee Membership</span>}
+                          hiddenContent={
+                            <Ul gap={4}>
+                              {nonCouncilMemberRoles[role.id][0].map((activeRole) => (
+                                <li key={activeRole.id}>
+                                  <strong>{`${activeRole.title}: `}</strong>
+                                  <Link
+                                    to={{
+                                      pathname: "/events",
+                                      state: {
+                                        committees: { [activeRole.body?.id as string]: true },
+                                      },
+                                    }}
+                                  >
+                                    <strong>{`${activeRole.body?.name}`}</strong>
+                                  </Link>
```

Comment:
Two things, this is duplicate code with the nonCouncilMemberRoles below this. May be worth it to make a generator function for this component so that if we want to update this in the future you only need to update the generator instead of two spots in the code.

Second, hot take and I know this is a _technically_ an accessibility violation but I think we links should always stay blue even after clicked / visited. Because we don't pass the body as a query parameter in the URL, when you click on any single one of these and then return back to the person page, all of them are now purple. Which imo, is a _worse_ experience than all always blue.

Reply:
1. sonds good
2. I saw the same problem, will add a ?body=id to the link

Reply:
I would be fine with adding a query parameter to specify a single body sure!